File: backend/Crystal.Core/Entities/LeaveRequest.cs
using Crystal.Core.Enums;

namespace Crystal.Core.Entities;

public class LeaveRequest
{
    public int Id { get; set; }

    public int EmployeeProfileId { get; set; }
    public EmployeeProfile EmployeeProfile { get; set; } = null!;

    public LeaveType LeaveType { get; set; }
    public LeaveRequestStatus Status { get; set; } = LeaveRequestStatus.Pending;

    public DateOnly StartDate { get; set; }
    public DateOnly EndDate { get; set; }
    public string? Reason { get; set; }

    public bool IsDeleted { get; set; } = false;
}

File: frontend/src/data/types/hr/leaveRequest.ts
export const LEAVE_TYPES = {
    Vacation: "Vacation",
    Sick: "Sick",
    Unpaid: "Unpaid",
    Other: "Other",
} as const;

export const LEAVE_REQUEST_STATUSES = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
} as const;

export type LeaveType = (typeof LEAVE_TYPES)[keyof typeof LEAVE_TYPES];
export type LeaveRequestStatus = (typeof LEAVE_REQUEST_STATUSES)[keyof typeof LEAVE_REQUEST_STATUSES];

export interface LeaveRequest {
    id: number;
    employeeProfileId: number;
    employeeFirstName: string;
    employeeLastName: string;
    leaveType: LeaveType;
    status: LeaveRequestStatus;
    startDate: string;
    endDate: string;
    reason: string | null;
    isDeleted: boolean;
}

export interface LeaveRequestApiDto {
    id: number;
    employeeProfileId: number;
    employeeFirstName: string;
    employeeLastName: string;
    leaveType: string;
    status: string;
    startDate: string;
    endDate: string;
    reason: string | null;
}

export interface LeaveRequestFormData {
    employeeProfileId: number;
    leaveType: LeaveType;
    startDate: string;
    endDate: string;
    reason: string | null;
}

export interface LeaveRequestStatusUpdatePayload {
    status: LeaveRequestStatus;
}
